### DISCUSSION TITLE
USB Ports on Devices Such as Servers and Network Switches

### DISCUSSION THREAD

**Discussion by iamk3** [netbox-community/netbox#17552] [Q&A]
What is the best method for accounting for USB ports on devices such as Servers, NAS, Network Switches, etc... I have several of these type where we are using the USB ports in a variety of ways and I'd really like to make sure those connections and such are documented in Netbox.

**Comment by DanSheps** [comment]
Short answer: don't unless the USB port is a console port (Cisco Catalyst, etc)

**Comment by ghost** [comment]
I agree. this would be challenging. USB ports aren't typically labeled in any meaningful way on servers. How are you sure that what you say is USB port 3 is what someone else looks at the back of the server and also considers USB port 3?

When you have 3 banks of 2 ports each .. which one is port 3? from the left? from the right? is it across? or stacked? very confusing.

**Comment by candlerb** [comment]
If it's a USB type A then I make it "console server port" and if it's USB type B (or mini B or micro B) then I make it "console port".  That's correct for the case of USB consoles, but should also work for other USB uses.

You can make USB hubs as devices with one console port and multiple console server ports.

Fortunately I've never had to deal with USB C in a data centre, but I guess you'd have to assign roles: console server port for the "PC / controller" side and console port for the "peripheral" side.

**Reply by sigprof** [reply]
> If it's a USB type A then I make it "console server port" and if it's USB type B (or mini B or micro B) then I make it "console port".

This looks correct for a case like UPS monitoring over USB.

However, if the server is connected to a KVM switch, the roles would be reversed — the KVM switch should probably be considered a “console server”, and the server itself should have a “console port”.  I guess that the simplest way to handle this case is to ignore the USB connection which provides the keyboard and mouse interface, and model just the video connection (so that the server would have a “console port” with the VGA connector, the KVM switch would have a number of “console server ports”, and the extra USB connections between the KVM switch and the servers would be implicit).  The USB port modeled as “console server port” could be manually marked as connected in this case (but it would be impossible to represent that connection as a cable in NetBox unless the KVM switch would have some fake “console” ports to match, which would probably be really confusing).
